src/globalvars.cpp

```diff
@@ -258,19 +307,110 @@ void GlobalVars::verify_maps_found(
     const std::unordered_map<std::string, struct bpf_map *>
         &section_name_to_global_vars_map)
 {
-  for (const auto &[name, config] : global_var_map_) {
+  for (const auto &[name, config] : added_global_vars_) {
     if (!section_name_to_global_vars_map.contains(config.section)) {
       LOG(BUG) << "No map found in " << config.section
                << " which is needed to set global variable " << name;
     }
   }
 }
 
+Result<GlobalVarMap> GlobalVars::get_named_param_vals(
+    std::vector<std::string> raw_named_params) const
+{
+  std::unordered_map<std::string, std::optional<std::string>> named_params;
+  std::vector<std::string> unexpected_params;
+
+  for (const auto &param : raw_named_params) {
+    if (param.find("=") != std::string::npos) {
+      auto split = util::split_string(param,
+                                      '=',
+                                      /* remove_empty= */ true);
+      named_params[split[0]] = split[1];
+      if (!added_global_vars_.contains(split[0])) {
+        unexpected_params.push_back(split[0]);
+      }
+    } else {
+      named_params[param] = std::nullopt;
+      if (!added_global_vars_.contains(param)) {
+        unexpected_params.push_back(param);
+      }
+    }
+  }
+
+  if (!unexpected_params.empty()) {
+    std::vector<std::string> expected_params;
+    for (const auto &[name, _] : named_param_defaults_) {
+      expected_params.push_back(name);
+    }
+
+    return make_error<UnknownParamError>(std::move(unexpected_params),
+                                         std::move(expected_params));
+  }
+
+  GlobalVarMap named_param_vals;
+
+  for (const auto &[name, default_val] : named_param_defaults_) {
+    const auto &config = added_global_vars_.at(name);
+    auto it = named_params.find(name);
+
+    // Use the default
+    if (it == named_params.end()) {
+      if (config.type == Type::integer) {
+        named_param_vals[name] = std::get<int64_t>(default_val);
+      } else if (config.type == Type::boolean) {
+        named_param_vals[name] = std::get<bool>(default_val);
+      } else if (config.type == Type::string) {
+        named_param_vals[name] = std::get<std::string>(default_val);
+      }
+      continue;
+    }
+
+    // No value means it's a boolean
+    if (!it->second.has_value()) {
+      if (config.type == Type::integer) {
+        return make_error<NamedParamError>(name, "true", "expects an integer");
+      } else if (config.type == Type::string) {
+        return make_error<NamedParamError>(name, "true", "expects a string");
+      } else if (config.type == Type::boolean) {
+        named_param_vals[name] = true;
+      }
+      continue;
+    }
+
+    const std::string &val = *it->second;
+
+    if (config.type == Type::integer) {
+      try {
+        named_param_vals[name] = std::stoll(val);
```

Comment:
Did you want to parse using the util::to_uint? (You'd need to explicitly check for negative and handle that case, but it's simple.) That way you could pass `--foo=43ns` for example.

This also lets you avoid the exception-based flow, which I think we should try to do whenever possible.

Reply:
Ack. I like it